index.bs

```diff
@@ -90,6 +100,37 @@ The exact type of the `globalScope` can vary across runtimes. Most Web Platform
 
 With many runtimes, adding a new global-scoped property can introduce breaking changes when the new global conflicts with existing application code. Many Web Platform APIs define global properties using the `readonly` attribute. To avoid introducing breaking changes, runtimes conforming to this specification MAY choose to ignore the `readonly` attribute for properties being added to the global scope.
 
+Requirements for EcmaScript modules {#esm-requirements}
+=======================================================
+
+[=Web-interoperable runtimes=] which support EcmaScript modules must implement the following ECMA-262 host hooks as follows:
+
+: [$HostLoadImportedModule$]
+:: This host hook must be implemented [[HTML#hostloadimportedmodule|as defined in the HTML spec]], except that they may pass a [=perform the fetch hook=] algorithm to [=fetch a single imported module script=] even when <var ignore="">loadState</var> is undefined. (See also [[#custom-module-loading]].)
+: [$HostGetSupportedImportAttributes$]
+:: This host hook must be implemented [[HTML#hostgetsupportedimportattributes|as defined in the HTML spec]].
+: [$HostGetImportMetaProperties$]
+:: If a runtime can never have an [=entrypoint module=] for any [=agent clusters=] (e.g. web browsers), then it may implement this host hook [[HTML#hostgetimportmetaproperties|as defined in the HTML spec]]. Otherwise, it must implement it as follows:
```

Comment:
If a runtime never has entry points, is there any observable difference between the HTML algorithm and the one below?

Reply:
With the algorithm below, `import.meta.main` will exist and be set to `false` for non-entrypoint modules, whereas with the HTML algorithm `import.meta.main` does not exist.